src/zenml/integrations/aws/secrets_manager/aws_secrets_manager.py

```diff
@@ -0,0 +1,162 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+from typing import Any, ClassVar, Dict, List, Optional
+
+import boto3 as boto3
+from botocore.exceptions import ClientError
+
+from zenml.enums import SecretsManagerFlavor, StackComponentType
+from zenml.logger import get_logger
+from zenml.secret_sets.base_secret_set import BaseSecretSet
+from zenml.secrets_manager.base_secrets_manager import BaseSecretsManager
+from zenml.stack.stack_component_class_registry import (
+    register_stack_component_class,
+)
+
+logger = get_logger(__name__)
+
+DEFAULT_AWS_REGION = "us-east-1"
+
+
+@register_stack_component_class(
+    component_type=StackComponentType.SECRETS_MANAGER,
+    component_flavor=SecretsManagerFlavor.AWS,
+)
+class AWSSecretsManager(BaseSecretsManager):
+    """Class to interact with the AWS secret manager."""
+
+    supports_local_execution: bool = True
+    supports_remote_execution: bool = True
+    region_name: str = DEFAULT_AWS_REGION
+
+    CLIENT: ClassVar[Any] = None
+
+    @classmethod
+    def _ensure_client_connected(cls, region_name: str):
```

Comment:
Another point here, does the client really maintain any connection state? or does it only connect to the api once you make calls? My impression from working with boto3 was that it doesn't actually do anything much when you do things like create session, or create clients (not to mention the frustrating fact that there are multiple different ways to create a client), so maybe you could just remove the class member entirely and simply call `boto3.client('secrets_manager')` or whatever everytime you use it.... it's not like it's keeping a database connection up, so it would make it easier on us not to manage that state